File: Python/ProgramStudi.py
from Course import Course
from Mahasiswa import Mahasiswa
from Dosen import Dosen


class ProgramStudi:
    __nama_prodi = ""
    __kode_prodi = ""
    __listcrs: Course = []
    __listmhs: Mahasiswa = []
    __listdosen: Dosen = []

    def __init__(self, nama_prodi="", kode_prodi=""):
        self.__nama_prodi = nama_prodi
        self.__kode_prodi = kode_prodi

    def set_nama_prodi(self, nama_prodi):
        self.__nama_prodi = nama_prodi

    def set_kode_prodi(self, kode_prodi):
        self.__kode_prodi = kode_prodi

    def get_nama_prodi(self):
        return self.__nama_prodi

    def get_kode_prodi(self):
        return self.__kode_prodi

    def add_course(self, course: Course):
        self.__listcrs.append(course)

    def add_mahasiswa(self, mahasiswa: Mahasiswa):
        self.__listmhs.append(mahasiswa)

    def add_dosen(self, dosen: Dosen):
        self.__listdosen.append(dosen)
        # self.__listdosen += dosen
        # pass

    def get_listcrs(self):
        return self.__listcrs

    def get_listmhs(self):
        return self.__listmhs

    def get_listdosen(self):
        return self.__listdosen

File: C++/ProgramStudi.cpp
#include "header.hh"
ProgramStudi::ProgramStudi()
{
    this->nama_prodi = "";
    this->kode_prodi = "";
    this->mahasiswa = {};
    this->dosen = {};
    this->course = {};
}
ProgramStudi::ProgramStudi(string nama_prodi, string kode_prodi)
{
    this->nama_prodi = nama_prodi;
    this->kode_prodi = kode_prodi;
    this->mahasiswa = mahasiswa;
    this->dosen = dosen;
    this->course = course;
}
void ProgramStudi::set_nama_prodi(string nama_prodi)
{
    this->nama_prodi = nama_prodi;
}
void ProgramStudi::set_kode_prodi(string kode_prodi)
{
    this->kode_prodi = kode_prodi;
}
void ProgramStudi::set_mahasiswa(vector<Mahasiswa> mahasiswa)
{
    this->mahasiswa = mahasiswa;
}
void ProgramStudi::set_dosen(vector<Dosen> dosen)
{
    this->dosen = dosen;
}
void ProgramStudi::set_course(vector<Course> course)
{
    this->course = course;
}
string ProgramStudi::get_nama_prodi()
{
    return this->nama_prodi;
}
string ProgramStudi::get_kode_prodi()
{
    return this->kode_prodi;
}
vector<Mahasiswa> ProgramStudi::get_mahasiswa()
{
    return this->mahasiswa;
}
vector<Dosen> ProgramStudi::get_dosen()
{
    return this->dosen;
}
vector<Course> ProgramStudi::get_course()
{
    return this->course;
}
void ProgramStudi::add_course(Course course)
{
    this->course.push_back(course);
}
void ProgramStudi::add_mahasiswa(Mahasiswa mahasiswa)
{
    this->mahasiswa.push_back(mahasiswa);
}
void ProgramStudi::add_dosen(Dosen dosen)
{
    this->dosen.push_back(dosen);
}
ProgramStudi::~ProgramStudi()
{
}